packages/react-components/src/gallery/dominantSpeaker.test.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { VideoGalleryParticipant } from '..';
+import { smartDominantSpeakerParticipants } from './dominantSpeaker';
+
+const participants: VideoGalleryParticipant[] = [
+  { userId: '1' },
+  { userId: '2' },
+  { userId: '3' },
+  { userId: '4' },
+  { userId: '5' },
+  { userId: '6' },
+  { userId: '7' },
+  { userId: '8' }
+];
+
+const dominantSpeakers: Array<string> = ['1', '3', '5', '7'];
+
+const visibleParticipants: VideoGalleryParticipant[] = [
+  { userId: '1' },
+  { userId: '2' },
+  { userId: '3' },
+  { userId: '4' }
+];
+
+describe('Test smartDominantSpeakerParticipants function', () => {
+  test('returns all participants when number of participants is less than or equal to number of dominant speakers', () => {
+    let result = smartDominantSpeakerParticipants(participants.slice(0, 3), dominantSpeakers, visibleParticipants);
+    expect(result.length).toBe(3);
+
+    result = smartDominantSpeakerParticipants(participants.slice(0, 4), dominantSpeakers, visibleParticipants);
+    expect(result.length).toBe(4);
+  });
+
+  test('returns no more than the MAX participants', () => {
+    let result = smartDominantSpeakerParticipants(participants, dominantSpeakers, visibleParticipants, 4);
+    expect(result.length).toBe(4);
+
+    result = smartDominantSpeakerParticipants(participants, dominantSpeakers, visibleParticipants, 5);
+    expect(result.length).toBe(5);
+  });
+
+  test('returns participants even when there are no visible participants', () => {
+    const result = smartDominantSpeakerParticipants(participants, dominantSpeakers, [], 4);
+    expect(result.length).toBe(4);
+  });
+
+  test('returns participants array replacing non-dominant speakers with dominant speakers', () => {
+    const result = smartDominantSpeakerParticipants(participants, dominantSpeakers, visibleParticipants, 4);
+    const resultUserIds = result.flatMap((p) => p.userId);
+    expect(resultUserIds).toEqual(['1', '5', '3', '7']);
+  });
+
+  test('maintains the ordering of new participants to match last visible participants', () => {
+    const result = smartDominantSpeakerParticipants(
+      participants,
+      ['1', '2', '3', '4'],
+      visibleParticipants.slice(2, 4),
```

Comment:
This is a bit of a matter of taste, but I like unittests to be DAMP (the opposite of Dont Repeat Yourself, but I forget what the letters stand for) -- I don't want to go understand what `visibleParticipants.slice(2, 4)` does. I'd just inline the exact object here. I'd actually also inline the `participants` list in each of these tests (perhaps making the list shorter. All these tests don't need 8 participants each. Just give them what they need, and no more).